pkg/dependencies/factory.go

```diff
@@ -667,7 +667,7 @@ func (f *Factory) WithAwsIamAuth() *Factory {
 		}
 		certgen := crypto.NewCertificateGenerator()
 		clusterId := uuid.New()
-		f.dependencies.AwsIamAuth = awsiamauth.NewAwsIamAuth(certgen, clusterId)
```

Comment:
I think we need to add this at the top of `WithAwsIamAuth`

```go
f.WithKubectl().WithWriter()
```

Reply:
Done